### PR TITLE
No unchecked indexed access

### PR DESCRIPTION
**Pull request by angrykoala**
# Description
Adds `"noUncheckedIndexedAccess": true` option in `tsconfig` for the library and cypher builder. This change makes typings on strings and records more strict.

A lot of small changes to comply with this stricter typings

## Complexity

Complexity: Medium

### CODE REVIEW

**Comment by neo4j-team-graphql** [comment]
Made some changes to the GraphQL Toolbox? Or made some changes to the core library perhaps?

Check out your changes at https://neo4j-graphql-toolbox-stage-3144.surge.sh

Use for instance a local dbms hosted on [Neo4j Desktop](https://neo4j.com/download/) or a cloud-hosted [Aura Free](https://neo4j.com/cloud/aura-free/) dbms.

File: packages/graphql/tests/integration/directives/fulltext/fulltext-query.int.test.ts [hunk 1]
```diff
@@ -2039,7 +2039,7 @@ describe("@fulltext directive", () => {
             try {
                 const result = await session.run(indexQueryCypher);
 
-                expect(result.records[0].get("result")).toEqual({
+                expect(result.records[0]?.get("result")).toEqual({
```
Comment (mjfwebb) [inline packages/graphql/tests/integration/directives/fulltext/fulltext-query.int.test.ts:2042]: This file is a good example of where we could use the generic param available to us in `session.run`

**Review by mjfwebb — COMMENTED** [review]
Some thoughts :)

**Review by mjfwebb — DISMISSED** [review]
Great start to a cleaner and more secure code base :)

**Pull request merged**